modules/input.nix

```diff
@@ -0,0 +1,104 @@
+{ config, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.plasma;
+  numlockSettings = [ "on" "off" "unchanged" ];
+in
+{
+  # Keyboard options
+  options.programs.plasma.input.keyboard = {
+    layouts = mkOption {
+      type = with types; nullOr (listOf str);
+      default = null;
+      example = [ "es" "us" ];
+      description = ''
+        Keyboard layouts to use.
+      '';
+    };
+    numlockOnStartup = mkOption {
+      type = with types; nullOr (enum numlockSettings);
+      default = null;
+      example = "on";
+      description = ''
+        Numpad settings at startup.
+      '';
+    };
+  };
+
+  config.programs.plasma.configFile."kxkbrc" = mkIf (cfg.enable) (
+    mkMerge [
+      (
+        mkIf (cfg.input.keyboard.layouts != null) {
+          Layout = {
+            Use.value = true;
+            LayoutList.value = strings.concatStringsSep "," cfg.input.keyboard.layouts;
+          };
+        }
+      )
+    ]
+  );
+
+  # Trackpad options
+
+  options.programs.plasma.input.touchpad = {
```

Comment:
The options here will write to the same group as mentioned in #47, am I right? If so we should probably add `DisableWhileTyping`, `NaturalScroll` and `TapToClick` as options which would close that issue.

Reply:
Yes!
I was going to add those first, but I thought that making the user look for the IDs of the trackpad, and then turning them to decimal was a bit too dumb. 

Already took care of the hexadecimal to decimal part here, but I'll do the "looking for the IDs" part in a separate pr, since thinking about how to do that is what burnt me out, and I don't want this pr to be stalled for another month :(

Reply:
No worries, I think just getting this into `plasma-manager` gets us a long way.
Looking for the IDs isn't necessarily going to be that easy, and would, if done at activation-time not be reproducible either (since the output would depend on the devices attached when the activation is applied). I think when doing that we could maybe consider just altering the config-file generated automatically by kde, which contains these ids aleady, in a startup-script, as that would be declarative, and maybe easier to do. I would maybe require a reload of some kde component, but `plasma-manager` should support that already.

Anyway that will be in the future, there's no need to think about this yet (and I think that it isn't necessarily a must for this module anyway).